File: api/pkg/apis/core/validation/validation.go
package validation

import (
	"github.com/nrc-no/core/api/pkg/apis/core"
	"k8s.io/apimachinery/pkg/util/validation/field"
)

func ValidateFormDefinition(f *core.FormDefinition) field.ErrorList {
	allErrs := field.ErrorList{}

	allErrs = append(allErrs, ValidateFormDefinitionSpec(&f.Spec, field.NewPath("spec"))...)

	return allErrs
}

func ValidateFormDefinitionSpec(f *core.FormDefinitionSpec, fldPath *field.Path) field.ErrorList {
	allErrs := field.ErrorList{}

	if len(f.Group) == 0 {
		allErrs = append(allErrs, field.Invalid(fldPath.Child("group"), f.Group, "cannot be set without group"))
	}

	return allErrs
}

File: ui/libs/shared/api-client/src/lib/api/meta/v1/validation.ts
import { TypeMeta } from './types';
import { Path } from '../../../field';
import { ErrorList, Required } from '../../../field';

export const validateTypeMeta = (typeMeta: TypeMeta, fldPath: Path): ErrorList => {
  const errorList: ErrorList = [];
  if (typeMeta.kind === '') {
    errorList.push(Required(fldPath.child('kind'), 'kind is required'));
  }
  if (!typeMeta.apiVersion) {
    errorList.push(Required(fldPath.child('apiVersion'), 'apiVersion is required'));
  }
  return errorList;
};
